sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -172,9 +174,10 @@ def get_pm_all(self):
         # When ceased by the host, releases the freeze request, allowing the 
         # reported minimum, maximum and average values to update again.
         self.xcvr_eeprom.write(consts.VDM_CONTROL, VDM_FREEZE)
-        time.sleep(1)
-        self.xcvr_eeprom.write(consts.VDM_CONTROL, VDM_UNFREEZE)
+        #Retry for freeze done status
         PM_dict = dict()
+        if self.util_vdm_freeze_unfreeze_status_retry(VDM_FREEZE_UNFREEZE_DONE_RETRY, SLEEP_50MS, consts.VDM_FREEZE_DONE, 1):
+            helper_logger.log_error('Freeze done status check timeout expired')
```

Comment:
@jaganbal since c_cmis.py is just an API, can we have separate APIs for freeze and unfreeze. ? The PM thread will have the logic to retry outside get_pm_all() and fetch the PM after freeze?

PM thread:-

1. Freeze VDM -> Separate API
2. get_pm_all
3. unfreeze VDM -> Separate API